src/pretix/control/views/item.py

```diff
@@ -428,13 +428,13 @@ def get_answer_statistics(self):
                 del a['options__answer']
         elif self.object.type in (Question.TYPE_TIME, Question.TYPE_DATE, Question.TYPE_DATETIME):
             qs = qs.order_by('answer')
-            qs_model = qs
-            qs = qs.values('answer').annotate(count=Count('id')).order_by('-count')
-            for a, a_model in zip(qs, qs_model):
+            model_cache = {a.answer: a for a in qs}
+            qs = qs.values('answer').annotate(count=Count('id'))
+            for a in qs:
                 a['alink'] = a['answer']
-                a['answer'] = str(a_model)
+                a['answer'] = str(model_cache[a['answer']])
         else:
-            qs = qs.order_by('answer').values('answer').annotate(count=Count('id')).order_by('-count')
+            qs = qs.values('answer').annotate(count=Count('id')).order_by('-count')
```

Comment:
Are you sure this change is necessary? Normally, order is relevant when dealing with GROUP BY clauses. Otherwise, the PR looks fine to me now. Probably we should just put ``.order_by()`` at the end of the queries, also for the datetimes. See also: https://docs.djangoproject.com/en/2.0/topics/db/aggregation/#interaction-with-default-ordering-or-order-by

Reply:
I don't think GROUP BY cares about the order, but we wouldn't get consistent ordering based just on the count.